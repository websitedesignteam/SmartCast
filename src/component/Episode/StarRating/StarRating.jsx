import React from 'react';
import { useRef, useState, useEffect } from 'react';
import styles from "./StarRating.module.scss";
import { yellow, gray } from "../../../utils/constants";

function Rating(props) {
  const [currentRating, setCurrentRating] = useState(0);
  const [data, setData] = useState([]);
  const ratingRef = useRef();
  const starRef = useRef([]);

  const hoverHandler = (event) => {
    const el = starRef.current;
    const hoverValue = event.target.dataset.value;
    Array.from(el).forEach(star => {
      star.style.color = hoverValue >= star.dataset.value ? yellow : gray;
    });
  };

  const setRating = (event) => {
    const el = starRef.current;
    if (!el) return;
    Array.from(el).forEach(star => {
    star.style.color =
        currentRating >= star.dataset.value ? yellow : gray;
    });
  };

  const starClickHandler = (event) => {
    const rating = event.target.dataset.value;
    setCurrentRating(rating); // set state so the rating stays highlighted
    if(props.onClick){
      props.onClick(rating); // emit the eventent up to the parent
    }
  };

  useEffect( () => {
    let data = ["star", "star", "star", "star", "star"];
    starRef.current = new Array(data.length);
    setData(data);

}, []);


  return (
    <div
      className={styles["rating"]}
      ref={ratingRef}
      data-rating={currentRating}
      onMouseOut={setRating}
    >
      {data.map((element, i) => 
          <span
            className={styles["star"]}
            ref={el => starRef.current[i] = el}
            key={i+1}
            data-value={i+1}
            onMouseOver={hoverHandler}
            onClick={starClickHandler}
          >
            &#9733;
          </span>
      )}
    </div>
  );
}

export default Rating;