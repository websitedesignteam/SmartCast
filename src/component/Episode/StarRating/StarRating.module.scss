@import "../../../styles/variables";

.rating {
    font-size: 1.5rem;
    color: $dark-gray;
}