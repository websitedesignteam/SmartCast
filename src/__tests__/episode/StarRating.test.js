import React from 'react';
import { shallow } from 'enzyme';
import StarRating from 'src/component/Episode/StarRating/StarRating';

test("should render the StarRating", () => {
    const onClick = () => {}
    const props = {
        onClick,

    }
    const component = shallow(<StarRating {...props} />);
    // expect(component.getElements()).toMatchSnapshot();
});