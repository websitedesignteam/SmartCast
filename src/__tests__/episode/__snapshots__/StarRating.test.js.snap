// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the StarRating 1`] = `
Array [
  <div
    className="rating"
    data-rating={0}
    onMouseOut={[Function]}
  />,
]
`;
